include/DDpackage.hpp

```diff
@@ -0,0 +1,2047 @@
+/*
+ * This file is part of the JKQ DD Package which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum_dd/ for more information.
+ */
+
+#ifndef DDpackage_H
+#define DDpackage_H
+
+#include "Complex.hpp"
+#include "ComplexCache.hpp"
+#include "ComplexNumbers.hpp"
+#include "ComplexTable.hpp"
+#include "ComplexValue.hpp"
+#include "ComputeTable.hpp"
+#include "Control.hpp"
+#include "Definitions.hpp"
+#include "Edge.hpp"
+#include "NoiseOperationTable.hpp"
+#include "ToffoliTable.hpp"
+#include "UnaryComputeTable.hpp"
+#include "UniqueTable.hpp"
+
+#include <array>
+#include <bitset>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <map>
+#include <regex>
+#include <set>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace dd {
+    class Package {
+        ///
+        /// Complex number handling
+        ///
+    public:
+        ComplexNumbers cn{};
+
+        ///
+        /// Construction, destruction, information and reset
+        ///
+    public:
+        static constexpr auto maxPossibleQubits = std::numeric_limits<Qubit>::max() + 1;
```

Comment:
This will be UB as soon as `Qubit` is typedef'ed as `int`.

I'd suggest the following:
```
static constexpr std::size_t maxPossibleQubits = static_cast<std::make_unsigned_t<Qubit>>(std::numeric_limits<Qubit>::max()) + 1u;
```

Reply:
Good point. Fixed 👍🏼